node/node_v2.go

```diff
@@ -0,0 +1,126 @@
+// These v2 methods are implemented in this separate file to keep the code organized.
+// Note that there is no NodeV2 type and these methods are implemented in the existing Node type.
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/gammazero/workerpool"
+)
+
+type requestMetadata struct {
+	blobShardIndex int
+	quorum         core.QuorumID
+}
+type relayRequest struct {
+	chunkRequests []*clients.ChunkRequestByRange
+	metadata      []*requestMetadata
+}
+type response struct {
+	metadata []*requestMetadata
+	bundles  [][]byte
+}
+
+type RawBundles struct {
+	BlobCertificate *corev2.BlobCertificate
+	Bundles         map[core.QuorumID][]byte
+}
+
+func (n *Node) DownloadBatch(ctx context.Context, batch *corev2.Batch) ([]*corev2.BlobShard, []*RawBundles, error) {
```

Comment:
It seems making robust retry logic for this method could become pretty complex. Are you thinking of doing this as a follow-on/post-MVP?

Reply:
I was thinking about not retrying at all (at least for the MVP) and returning error immediately if download fails. What do you think?